pkg/pex/gossip.go

```diff
@@ -63,12 +57,31 @@ type GossipConfig struct {
 	MaxMsgSize uint64
 }
 
-func (g GossipConfig) tail() func(View) View {
-	return tail(g.Protect)
-}
-
-func (g GossipConfig) head() func(View) View {
-	return head((g.MaxView / 2) - 1)
+func (g *GossipConfig) applyDefaults() {
```

Comment:
Debatable opinion:  it is better to panic if the user-provided function passed to `WithGossip` sets an invalid value.  Currently, the cluster will be implicitly configured to behave in a manner that the user did not expect.  I think it's better to fail noisily here.  Thoughts?

Moreover, I think most of these values are invalid if they are `<= 0`, rather than simply negative.  We might consider making each of these a `uint` and panicking `g.XXX == 0`.

This of course means the zero-value is no longer "ready to use", but I think that's fine.  I had initially planned to pass a concrete `GossipConfig` to `WithGossip`, but then reverted back to a factory function in order to support per-namespace configuration.  So, I don't think there's much value in allowing a zero-value struct anymore.

Reply:
I think you are right and we should make them `uint`. However, 0 values cannot be invalid, as it should be possible to set protect or swap to zero if wanted. Therefore, I think we should simply make them `uint` and don't apply defaults.

Also, I will create a new variable which is `DefaultGossipConfig`, so that users can modify easier single fields such as `timeout`, or `Tick`.